ee/query-service/app/server.go

```diff
@@ -199,17 +202,61 @@ func (s *Server) createPrivateServer(apiHandler *api.APIHandler) (*http.Server,
 	}, nil
 }
 
+func getPATToken(r *http.Request) (string, error) {
+	patHeader := r.Header.Get("SIGNOZ-API-KEY")
+	if patHeader == "" {
+		return "", nil
+	}
+
+	authHeaderParts := strings.Fields(patHeader)
+	if len(authHeaderParts) != 2 || strings.ToLower(authHeaderParts[0]) != "bearer" {
+		return "", fmt.Errorf("PAT authorization header format must be bearer {token}")
+	}
+
+	return authHeaderParts[1], nil
+}
+
 func (s *Server) createPublicServer(apiHandler *api.APIHandler) (*http.Server, error) {
 
 	r := mux.NewRouter()
 
+	getUserFromPAT := func(r *http.Request) (*model.UserPayload, error) {
+		patToken, err := getPATToken(r)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get PAT token in request headers, err: %v", err)
+		}
+		ctx := context.Background()
+		dao := apiHandler.AppDao()
+		pat, err := dao.GetPAT(ctx, patToken)
```

Comment:
possible optimization (optional): we could fetch user at the same time we validate pat. To reduce additional DB call.

Reply:
Done. Nice observation.